CMakeLists.txt

```diff
@@ -185,8 +185,12 @@ if(OPENMP_ENABLE)
   #message(STATUS "****** finding OpenMP support")
   FIND_PACKAGE(OpenMP)
   if(OPENMP_FOUND)
-    add_compile_options($<$<COMPILE_LANGUAGE:C>:${OpenMP_C_FLAGS}> $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>)
-    link_libraries(-fopenmp)
+    if(${CMAKE_VERSION} VERSION_GREATER "3.8.9")
```

Comment:
IS `VERSION_GREATER` really version greater or equal  
basically this is only on cmake 3.9 and above  so is 3.9.0 > 3.9?  if so I think you could put 3.9 here instead of some made up version number.

Reply:
One of their test cases has `message(SEND_ERROR "1.2 VERSION_EQUAL 1.2.0.0 is not true!")`, so I think 3.9 is equivalent to 3.9.0

Reply:
The way they advance the pointers for the version comparison seem a bit odd, like there could be cases where the version component on one side gets compared to the previous version component of the other. https://github.com/Kitware/CMake/blob/38e4246f37b832b88437e83710348c174be02508/Source/cmSystemTools.cxx#L2594

Reply:
It looks like it stops the comparison based on whichever version provides the least components. https://github.com/Kitware/CMake/blob/38e4246f37b832b88437e83710348c174be02508/Source/cmSystemTools.cxx#L2594

Reply:
ok, either way since we want 3.9, any statement with  VERSION_GREATER should start with 3.8 and if we have a patch component it has to be greater than 6 since that is the current version of 3.8.X 

I would probably change it to 3.8 in that case

Reply:
Alright, I have something that works now -- I misread what the loop is doing, it doesn't stop as soon as it hits the end of one version string.